### PR TITLE
Fix possible problems when cross compiling

### PR DESCRIPTION
**Pull request by AdrianFreundQC**
See https://github.com/conda-forge/staged-recipes/pull/26590#discussion_r1631341584

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by AdrianFreundQC** [comment]
@conda-forge-admin, please rerender

File: recipe/build.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-if [[ "${build_platform}" == linux-* ]]; then
+if [[ "${target_platform}" == linux-* ]]; then
     export OPENSSL_DIR=$PREFIX
 fi
```
Comment (pavelzw) [inline recipe/build.sh:5]: actually, let's remove this completely and add `pkg-config` to the build dependencies instead

**Review by pavelzw — CHANGES_REQUESTED** [review]

File: recipe/build.sh (hunk 1 above)
Comment (AdrianFreundQC) [reply recipe/build.sh:5]: I remember that to not work, but I'm not sure why or if my memory is correct.

Comment (pavelzw) [reply recipe/build.sh:5]: when i checked last time at another project, adding `pkg-config` fixed it as well

**Pull request closed without merging**